src/certifier.cc

```diff
@@ -116,7 +120,7 @@ int certifier::framework::policy_store::get_num_trusted_services() {
 }
 
 const trusted_service_message* certifier::framework::policy_store::get_trusted_service_info_by_index(int n) {
-  if (n >= num_ts_)
+  if ((n < 0) || (n >= num_ts_))
```

Comment:
You will see a collection of such changes to `_by_index()` interfaces, to catch illegal index IDs.

These were an initial set of fixes I ended up with ... and then held back. (There are other bugs in this file, but those may need to be taken up separately.)